### ISSUE TITLE
Adaptive Quadtrees (paper)

### ISSUE THREAD

**Issue by boberfly** [PixarAnimationStudios/OpenSubdiv#914]
Apologies if this is already in the codebase or not so practical for OSD's use case:
http://www.graphics.stanford.edu/~niessner/papers/2016/4subdiv/brainerd2016efficient.pdf

I'm curious about this paper in relation to OSD, is there a plan to have an implementation similar to this? Or if anyone has a good idea on how to implement this. Cheers!

**Comment by barfowl** [comment]
Yes, there is interest in this approach and there has been communication between the co-authors and the OpenSubdiv team (beginning prior to publication).  Priorities for 2016 unfortunately needed to address other areas, so we hope to dig deeper into this in 2017.

**Comment by jtran56** [comment]
Filed as internal issue #142045.

**Comment by boberfly** [comment]
Just thought I'd link this here for more info for others (last section of the video):
http://www.gdcvault.com/play/1024353/

**Comment by manuelk** [comment]
Working on it - pull requests are in the pipeline, but we are hitting some hardware / API limitations that were not immediately visible with the prototype code from the paper.

**Comment by boberfly** [comment]
Thanks for the update @manuelk

**Comment by boberfly** [comment]
Hi all, has there been any progress on this? I did notice a patch builder but it was more or less scaffolding for upcoming changes I'm guessing? Cheers.
